### PR TITLE
[TACHYON-1154] Improve JavaDoc and style of Journal Entries

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses: https://tachyon.atlassian.net/browse/TACHYON-1154?filter=-2

### CODE REVIEW

File: servers/src/main/java/tachyon/master/file/FileSystemMaster.java [hunk 1]
```diff
@@ -159,7 +159,7 @@ public void processJournalEntry(JournalEntry entry) throws IOException {
       PersistDirectoryEntry typedEntry = (PersistDirectoryEntry) entry;
       try {
         Inode inode = mInodeTree.getInodeById(typedEntry.getId());
-        inode.setPersisted(typedEntry.isPersisted());
+        inode.setPersisted(true);
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/FileSystemMaster.java]: Why always true?

Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/FileSystemMaster.java]: nvm

File: servers/src/main/java/tachyon/master/file/journal/DeleteFileEntry.java [hunk 2]
```diff
@@ -22,37 +22,57 @@
...
+   * Creates a new instance of {@link DeleteFileEntry}.
+   *
+   * @param id the id
+   * @param recursive the recursive flag
+   * @param opTimeMs the operation time (in milliseconds)
+   */
   @JsonCreator
-  public DeleteFileEntry(@JsonProperty("fileId") long fileId,
+  public DeleteFileEntry(
+      @JsonProperty("id") long id,
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/journal/DeleteFileEntry.java]: Note: this rename breaks the back-compatibility.

File: servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java [hunk 3]
```diff
@@ -52,11 +78,22 @@ public InodeFileEntry(@JsonProperty("creationTimeMs") long creationTimeMs,
...
-        new InodeFile.Builder().setName(mName).setBlockContainerId(BlockId.getContainerId(mId))
-            .setParentId(mParentId).setBlockSizeBytes(mBlockSizeBytes)
-            .setCreationTimeMs(mCreationTimeMs).setTTL(mTTL).setPersisted(mPersisted).build();
+        new InodeFile.Builder()
+            .setName(mName)
+            .setBlockContainerId(BlockId.getContainerId(mId))
+            .setParentId(mParentId)
+            .setBlockSizeBytes(mBlockSizeBytes)
+            .setCreationTimeMs(mCreationTimeMs)
+            .setTTL(mTTL)
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java]: There's `setTTL` method on `inodeFile`, do we still `setTTL` in the builder?

File: servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java [hunk 4]
```diff
@@ -103,7 +135,12 @@ public boolean isCacheable() {
   }
 
   @JsonGetter
-  public long getTtl() {
+  public long getTTL() {
     return mTTL;
   }
+
+  @Override
+  public JournalEntryType getType() {
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java]: If you're sorting the methods, could you also sort `getCacheable` and `getCompleted`?

File: servers/src/main/java/tachyon/master/file/journal/ReinitializeFileEntry.java [hunk 5]
```diff
@@ -36,16 +46,25 @@ public ReinitializeFileEntry(@JsonProperty("path") String path,
...
+  /**
+   * @return the block size (in bytes)
+   */
   @JsonGetter
   public long getBlockSizeBytes() {
     return mBlockSizeBytes;
   }
 
+  /**
+   * @return the TLL
```
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/journal/ReinitializeFileEntry.java]: TLL -> TTL

**Comment by yupeng9** [comment]
Thanks for the Javadoc improvement. I left some minor comments.

File: servers/src/main/java/tachyon/master/file/journal/DeleteFileEntry.java (hunk 2 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/journal/DeleteFileEntry.java]: Ack.

File: servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java (hunk 3 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java]: Is the idea that builder should only set options that are immutable?

File: servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java (hunk 4 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java]: I was not really sorting the getters, but for consistency across different journal types, I moved the getType() to the bottom. I can go ahead and sort all getters alphabetically if you'd prefer that.

File: servers/src/main/java/tachyon/master/file/journal/ReinitializeFileEntry.java (hunk 5 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/journal/ReinitializeFileEntry.java]: Done.

File: servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java (hunk 3 above)
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java]: oh, I just found it inconsistent: mPersisted, mPinned etc do not have setter in the builder

File: servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java (hunk 4 above)
Comment (yupeng9) [inline servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java]: That's fine. It's not enforced in our codebase

Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java]: Ack.

File: servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java (hunk 3 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java]: The convention I follow is as follows: For fields that can be set at object creation time, I use a build setter. For fields that can be set later, I use the regular setter. Thus, fields that can be set both at object creation time and later will have both the build setter and a regular setter.

Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/journal/InodeFileEntry.java]: I cleaned this up for InodeFile and InodeDirectory.

**Comment by jsimsa** [comment]
@yupeng9 addressed your comments, PTAL

**Comment by jsimsa** [comment]
Jenkins, test this please.

**Comment by jsimsa** [comment]
Jenkins, test this please.

**Comment by yupeng9** [comment]
LGTM

**Comment by jsimsa** [comment]
@uronce-cc @gpang Could you please take a look? Thanks.

File: servers/src/main/java/tachyon/master/block/journal/BlockContainerIdGeneratorEntry.java [hunk 6]
```diff
@@ -23,17 +23,25 @@
...
 
+  /**
+   * Creates a new instance of {@link BlockContainerIdGeneratorEntry}.
+   *
+   * @param nextContainerId the next container id
+   */
   @JsonCreator
-  public BlockContainerIdGeneratorEntry(@JsonProperty("nextContainerId") long nextContainerId) {
+  public BlockContainerIdGeneratorEntry(
+      @JsonProperty("nextContainerId") long nextContainerId) {
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/master/block/journal/BlockContainerIdGeneratorEntry.java]: This seems to deviate from our style? Should we add a new rule for this kind of argument list?

File: servers/src/main/java/tachyon/master/file/journal/AddMountPointEntry.java [hunk 7]
```diff
@@ -25,20 +25,21 @@
...
- * This class represents a journal entry for recording the addition of a new mount point.
+ * This class represents a journal entry for addition of a new mount point.
  */
 public class AddMountPointEntry extends JournalEntry {
   private final String mTachyonPath;
   private final String mUfsPath;
 
   /**
-   * Creates a new instance of <code>AddMountPointEntry</code>.
+   * Creates a new instance of {@link AddMountPointEntry}.
```
Comment (calvinjia) [inline servers/src/main/java/tachyon/master/file/journal/AddMountPointEntry.java]: Is it good practice to link back to the current class?

**Comment by uronce-cc** [comment]
`mvn_site.log` seems to be unrelated?

File: servers/src/main/java/tachyon/master/block/journal/BlockContainerIdGeneratorEntry.java (hunk 6 above)
Comment (uronce-cc) [inline servers/src/main/java/tachyon/master/block/journal/BlockContainerIdGeneratorEntry.java]: Yes, this style is more readable in this case.

File: servers/src/main/java/tachyon/master/file/journal/ReinitializeFileEntry.java [hunk 8]
```diff
@@ -23,11 +23,21 @@
...
 
+  /**
+   * Creates a new instance of {@link ReinitializeFileEntry}.
+   *
+   * @param path the path
+   * @param blockSizeBytes the block size (in bytes)
+   * @param ttl the TTL
+   */
   @JsonCreator
   public ReinitializeFileEntry(@JsonProperty("path") String path,
```
Comment (uronce-cc) [inline servers/src/main/java/tachyon/master/file/journal/ReinitializeFileEntry.java]: Why not keep the same style for `JsonProperty`as other entries?

File: servers/src/main/java/tachyon/master/file/meta/InodeDirectory.java [hunk 9]
```diff
@@ -66,6 +67,7 @@ public Object getFieldValue(Inode o) {
     }
   };
   @SuppressWarnings("unchecked")
+
```
Comment (uronce-cc) [inline servers/src/main/java/tachyon/master/file/meta/InodeDirectory.java]: Is this blank line intended?

File: servers/src/main/java/tachyon/master/block/journal/BlockContainerIdGeneratorEntry.java (hunk 6 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/block/journal/BlockContainerIdGeneratorEntry.java]: For the Entry constructors, I prefer to put each property on a new line if that's okay with you.

File: servers/src/main/java/tachyon/master/file/FileSystemMaster.java (hunk 1 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/FileSystemMaster.java]: Ack.

File: servers/src/main/java/tachyon/master/file/journal/AddMountPointEntry.java (hunk 7 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/journal/AddMountPointEntry.java]: I try to use link any time I refer to a Class or a Class member irrespective of whether it is a current class or not; hyperlinks FTW.

File: servers/src/main/java/tachyon/master/file/meta/InodeDirectory.java (hunk 9 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/meta/InodeDirectory.java]: Fixed.

**Comment by jsimsa** [comment]
Addressed your comments. PTAL

File: servers/src/main/java/tachyon/master/file/journal/ReinitializeFileEntry.java (hunk 8 above)
Comment (jsimsa) [inline servers/src/main/java/tachyon/master/file/journal/ReinitializeFileEntry.java]: Done.

**Comment by gpang** [comment]
LGTM

**Pull request merged**